notify/impl.go

```diff
@@ -153,20 +155,24 @@ func (w *Webhook) Notify(ctx context.Context, alerts ...*types.Alert) error {
 
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return err
+		return false, err
 	}
 
 	resp, err := ctxhttp.Post(ctx, http.DefaultClient, w.URL, contentTypeJSON, &buf)
 	if err != nil {
-		return err
+		return false, err
```

Comment:
I went through the code chain again. This is mostly not recoverable (malformed requests etc.) – however, there are some points like the server returning an HTTP response to an HTTPS request etc., that could in theory be resolved on the side of the integration.

I'd vote to make it retry until someone reports a problem. Inspecting the error for further differentiation is probably not worth the effort. Retrying is not exactly costly for us.